### DISCUSSION TITLE
如何使得import styles from  'xxx.less' 这样的格式也具备css module能力，import 'xxx.less' 这样就不需要cssmodule能力

### DISCUSSION THREAD

**Discussion by ChenXplorer** [web-infra-dev/rsbuild#2591] [Q&A]
如何使得import styles from  'xxx.less' 这样的格式也具备css module能力，import 'xxx.less' 这样就不需要cssmodule能力 ，目前看来是默认 xxx.moule.less 这样的具备 css module能力，另外可以自定义格式，但import styles from  'xxx.less'  不知道该如何配置

**Comment by chenjiahan** [comment]
This is not supported. We recommend using `foo.module.css` to name CSS Modules.

**Comment by ChenXplorer** [comment]
我现在是使用rsbuild 迁移了一个老的项目，当前是rsbuild 0.6.6版本，我进行了如下配置
` 
tools: {
          rspack: (rsconfig, { mergeConfig }) => {
            const res = mergeConfig(rsconfig, {
              module: {
                rules: [
                  {
                    test: /\.(js|ts|jsx|tsx)$/,
                    loader: require.resolve('./rspackCssmoduleLoader.js'),
                    exclude: [...exclude],
                    include: [new RegExp(srcPath), ...include],
                  },
                ],
              },
            });
            const cssRulesArr = [
              /\.css$/.toString(),
              /\.less$/.toString(),
              /\.s(a|c)ss$/.toString(),
            ];
            const rules =
              res?.module?.rules?.map((v: any) => {
                if (cssRulesArr.includes(v.test?.toString())) {
                  v.oneOf.unshift({
                    sideEffects: true,
                    resolve: { preferRelative: true },
                    use: v.oneOf?.[0].use,
                    resourceQuery: /\?module/, // 包含?module查询参数
                    type: 'css/module',
                  });
                }
                return v;
              }) || null;
            if (rules) {
              res.module!.rules = rules;
            }
            return res;
          },
        },
`

/rspackCssmoduleLoader.js 代码如下
`
const modifiedString = (inputString) => {
  // 匹配 import A from 'B.less'、import C from 'C.css' 和 import D from 'E.scss' 的模式，
  // 匹配 import A from "B.less"、import C from "C.css" 和 import D from "E.scss" 的模式，
  // 并替换为 import A from 'B.less?module'、import C from 'C.css?module' 和 import D from 'E.scss?module'

  const res = inputString.replace(
    /import\s+(\w+)?\s+from\s+(['"])([^'"]+\.(?:less|css|scss))\2/g,
    "import $1 from '$3?module'",
  );

  return res;
};

module.exports = function (source) {
  const modifySource = modifiedString(source);

  return modifySource;
};
`
进行了如上的配置是可以成功完成我的需求，但现在升级到0.7.6之后，无法成功运行项目
提示 Cannot read properties of undefined (reading 'unshift')  oneOf

打断点看应该是module rules 配置发生了改变，现在我不知道怎么才能基于现状进行修改，使得我的需求得以实现

**Reply by 9aoy** [reply]
0.7.0 里 css 规则有调整，可以开启 debug 模式，看下最新的配置结构。
https://rsbuild.dev/zh/community/releases/v0-7#%E6%9B%B4%E5%A5%BD%E7%9A%84-css-%E6%94%AF%E6%8C%81

https://rsbuild.dev/zh/guide/debug/debug-mode

**Comment by ChenXplorer** [comment]
done
